sycl/include/CL/sycl/INTEL/ac_int.hpp

```diff
@@ -0,0 +1,2458 @@
+    return ac_int<W2, S2>(op2).operator REL_OP(op);                            \
+  }
+
+#define ASSIGN_OP_WITH_INT(ASSIGN_OP, C_TYPE, W2, S2)                          \
+  template <int W, bool S>                                                     \
+  constexpr inline ac_int<W, S> &operator ASSIGN_OP(ac_int<W, S> &op,          \
+                                                    C_TYPE op2) {              \
+    return op.operator ASSIGN_OP(ac_int<W2, S2>(op2));                         \
+  }
+
+#define OPS_WITH_INT(C_TYPE, WI, SI)                                           \
+  BIN_OP_WITH_INT(*, C_TYPE, WI, SI, mult)                                     \
+  BIN_OP_WITH_INT(+, C_TYPE, WI, SI, plus)                                     \
+  BIN_OP_WITH_INT(-, C_TYPE, WI, SI, minus)                                    \
+  BIN_OP_WITH_INT(/, C_TYPE, WI, SI, div)                                      \
+  BIN_OP_WITH_INT(%, C_TYPE, WI, SI, mod)                                      \
+  BIN_OP_WITH_INT(>>, C_TYPE, WI, SI, arg1)                                    \
+  BIN_OP_WITH_INT(<<, C_TYPE, WI, SI, arg1)                                    \
+  BIN_OP_WITH_INT(&, C_TYPE, WI, SI, logic)                                    \
+  BIN_OP_WITH_INT(|, C_TYPE, WI, SI, logic)                                    \
+  BIN_OP_WITH_INT(^, C_TYPE, WI, SI, logic)                                    \
+                                                                               \
+  REL_OP_WITH_INT(==, C_TYPE, WI, SI)                                          \
+  REL_OP_WITH_INT(!=, C_TYPE, WI, SI)                                          \
+  REL_OP_WITH_INT(>, C_TYPE, WI, SI)                                           \
+  REL_OP_WITH_INT(>=, C_TYPE, WI, SI)                                          \
+  REL_OP_WITH_INT(<, C_TYPE, WI, SI)                                           \
+  REL_OP_WITH_INT(<=, C_TYPE, WI, SI)                                          \
+                                                                               \
+  ASSIGN_OP_WITH_INT(+=, C_TYPE, WI, SI)                                       \
+  ASSIGN_OP_WITH_INT(-=, C_TYPE, WI, SI)                                       \
+  ASSIGN_OP_WITH_INT(*=, C_TYPE, WI, SI)                                       \
+  ASSIGN_OP_WITH_INT(/=, C_TYPE, WI, SI)                                       \
+  ASSIGN_OP_WITH_INT(%=, C_TYPE, WI, SI)                                       \
+  ASSIGN_OP_WITH_INT(>>=, C_TYPE, WI, SI)                                      \
+  ASSIGN_OP_WITH_INT(<<=, C_TYPE, WI, SI)                                      \
+  ASSIGN_OP_WITH_INT(&=, C_TYPE, WI, SI)                                       \
+  ASSIGN_OP_WITH_INT(|=, C_TYPE, WI, SI)                                       \
+  ASSIGN_OP_WITH_INT(^=, C_TYPE, WI, SI)
+
+// ------------------------------------- End of Macros for Binary Operators with
```

Comment:
This a following comments shall be aligned

Reply:
Done. Thanks!

Reply:
All the macros should have a more defensive name, such as `__DPCPP_ASSIGN_OP_WITH_INT` or whatever.